tt_metal/common/test_tiles.hpp

```diff
@@ -353,3 +334,23 @@ inline std::vector<T> convert_layout(
     }
     return std::vector<T>();
 }
+
+template <typename T, template <typename...> typename BufferType>
+inline std::vector<T> convert_layout(
+    const BufferType<T>& inp,
+    tt::stl::Span<const uint32_t> shape,
+    tests::utils::TensorLayoutType inL,
+    tests::utils::TensorLayoutType outL,
+    std::optional<PhysicalSize> tile_shape = std::nullopt,
+    std::optional<PhysicalSize> face_shape = std::nullopt,
+    const bool transpose_within_face = false,
+    const bool transpose_of_faces = false) {
+    ZoneScoped;
+
+    TT_ASSERT(shape.size() >= 2, "Shape size {} must be at least rank 2!", shape.size());
+    uint32_t H = shape[shape.size() - 2], W = shape[shape.size() - 1];
```

Comment:
are you sure uint32_t here is enough?

Reply:
absolutely. We don't have enough memory to store a tensor that large